src/main/java/io/reactivex/Scheduler.java

```diff
@@ -106,13 +106,10 @@ public Disposable schedule(Runnable run) {
         }
         
         public Disposable schedulePeriodically(Runnable run, final long initialDelay, final long period, final TimeUnit unit) {
-            final SerialDisposable first = new SerialDisposable();
-
-            final SerialDisposable sd = new SerialDisposable(first);
-            
             final Runnable decoratedRun = RxJavaPlugins.onSchedule(run);
-            
-            first.replace(schedule(new Runnable() {
+
+            final SerialDisposable sd = new SerialDisposable();
```

Comment:
You can combine the two into a single class but it will cost 2 AtomicReferences either way:

```java
class TwoDisposable extends AtomicReference<Disposable> implements Disposable {
   final AtomicReference<Disposable> first;

   @Override
   public void replaceFirst(Disposable d) {
       DisposableHelper.replace(first, d);
   }

   public void replaceSecond(Disposable d) {
       DisposableHelper.replace(this, d);
   }

   @Override
   public void dispose() {
       DisposableHelper.dispose(this); 
       DisposableHelper.dispose(second);
   }
}
```